api/open-positions.ts

```diff
@@ -0,0 +1,47 @@
+import Airtable from "airtable";
+import { NowRequest, NowResponse } from "@now/node";
+
+interface JobRecord {
+  "Stav": "Aktivní" | "Obsazeno";
+  "Úkol/Role": string;
+  "Odkaz na celou příležitost": string | null;
+  "Detaily": string | null;
+  "Časová náročnost": string | null;
+  "Urgentní poptávka": boolean;
+}
+
+interface Position {
+  title: string;
+  isOpen: boolean;
+  isUrgent: boolean;
+  detailUrl: string | null;
+  description: string | null;
+  timeRequirements: string | null;
+}
+
+export default async (req: NowRequest, res: NowResponse) => {
+  try {
+    const apiKey = process.env.AIRTABLE_API_KEY as string;
+    const table = new Airtable({ apiKey }).base("apppZX1QC3fl1RTBM")(
+      "Hledané role a úkoly - To Be Moved"
+    );
+    const records = await table.select({ view: "Otevřené role" }).all();
+    const out = JSON.stringify(records.map(parseRecord), null, 2);
+    res.setHeader("Content-Type", "application/json");
+    res.status(200).send(out);
+  } catch (e) {
+    res.status(500).send(`Error: ${e}`);
```

Comment:
Raději bych interní chybu nevracel a jenom zalogoval, pokud máme nějaký log. Může to být nebezpečné.

Reply:
Když napíšete, co mám udělat místo toho, udělám.

Reply:
Ha, fér, díky. Navrhuju tam dát jen pětistovku a `Error: Pull requests welcome ;)`, což? :) Protože žádný vhodný stávající log mě nenapadá.

Reply:
> Function logs provide logging output for your Serverless Functions, this includes user introduced logging such as console.log, along with details of the request made to the function.

Stačí, když to vypíšeme a mělo by to pak jít najít ve Vercelu v logu.

Reply:
Ten log ve Vercelu je standardně realtime bez záznamu, museli bychom [zapínat nějaký log drain](https://vercel.com/integrations?category=logging). Navrhuju to skočit – když se to rozbije, můžem snadno diagnostikovat u sebe a kdyby se ukázalo, že nám logování chybí, můžem ho dodělat později?

Reply:
Přidal jsem `console.error()`, aby se to dalo ladit lokálně nebo i realtime na produkci a vracím 500 a JSON s generickou hláškou, aby neutekly nějaké tokeny apod. Za mě takhle dobrý.

Reply:
OK, supr. Akorát bych u pětistovky nevracel JSON – ukaž :)

Reply:
Tady: e821a50

Reply:
Jako mohl by to být vnd.error nebo ještě lépe [problem+json](https://tools.ietf.org/html/rfc7807), ale nechtěl jsem do toho moc zabrušovat.

Reply:
Já bych to právě nekomplikoval nad plain text nebo HTML, klient u té pětistovky stejně nemůže nic moc očekávat?

Reply:
Netrvám na tom. Je to jedno a v tomto případně dvakrát jedno. U vážně míněných API je dobré, když v případě chyby API vrací něco, co je klient schopen zpracovat, takže pokud je to API založené na JSONu, je dobré aby najednou klient nemusel parsovat důvod chyby z HTML a aby to bylo konzistentní. Je pak fajn, když API vrací třeba něco jako problem+json. Ale jestli vrací HTML nebo náhodný JSON s náhodnou strukturou, nota bene na pětikilo, je asi nepodstatné.

Reply:
Souhlas!, ale když vracíme obyčejnou hlášku, ze které stejně nic nevyparsuje, tak mně to přijde zbytečný.